### PR TITLE
Convert zstd-feedstock to v1 feedstock

### PR DESCRIPTION
**Pull request by baszalmstra**
This PR converts zstd-feedstock to a v1 recipe and switch the conda build tool to rattler-build.
It has been automatically generated with [feedrattler v0.3.14](https://github.com/hadim/feedrattler).

Changes:
- [x] 📝 Converted `meta.yaml` to `recipe.yaml`
- [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
- [x] 🔧 Updated `conda-forge.yml` to use `rattler-build` and `pixi` (optional)
- [x] 🔢 Bumped the build number
- [x] 🐍 Applied temporary fixes for `python_min` and `python_version`
- [x] 🔄 Rerender the feedstock with conda-smithy
- [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
:/ i am still on the fence about these v1 recipes.

I routinely use 

```
python build-locally.py --debug
```
and that still doesn't work with rattler build.

I won't object to some1 else pressing merge, but I don't see an strict advantage to move to rattler here, especially seeing as I loose an ability to debug packaging in an isolated way.

**Comment by baszalmstra** [comment]
Does https://rattler-build.prefix.dev/latest/debugging_builds/ help with that?

**Comment by hmaarrfk** [comment]
yes that seems to include the right information, but i would like to see it moved to conda-forge's build-locally.py 

Today, checking out this branch and running:
```
python build-locally.py --debug ...
+ echo 'rattler-build currently doesn'\''t support debug mode'
```

WIth the v0, it would fail because it is multi-ouptut, but i would generate delete all but the main output, and it would go through and drop me in a debug shell.

Ultimately, tensorflow and pytorch are multi-hour builds, so the ability to drop in a docker container with the right cuda versions installed is critical.

**Comment by hmaarrfk** [comment]
(the documentation you pointed to is really nice, and will read through it the next time i need to debug a rattler recipe)

**Comment by baszalmstra** [comment]
> Ultimately, tensorflow and pytorch are multi-hour builds, so the ability to drop in a docker container with the right cuda versions installed is critical.

Yeah, it makes sense that we improve that situation!

**Comment by hmaarrfk** [comment]
I understand the need to "test" feedrattler, do you think the "fixes" you had to apply will eventually be included in the tool? 

if so, perhaps we can try again once the fixes are in? 

if the fixes are "out of scope" for the automatic conversion, i'm more open to merging this.

**Comment by baszalmstra** [comment]
I hope those will be included in the tool yeah! But Im not the author of the tool. Im just a drive-by converter. 😅 Feel free to close this if you don't feel comfortable maintaining this.

**Comment by wolfv** [comment]
@hmaarrfk we've had the `rattler-build debug` feature since a while? What's missing? Is it just that the Python script is not up to date?

**Comment by hmaarrfk** [comment]
>  Is it just that the Python script is not up to date?

yes
